examples/pennylane_run_qaoa_maxcut.py

```diff
@@ -0,0 +1,288 @@
+        qml.RX(2 * beta, wires=wire)
+
+
+# unitary operator U_C with parameter gamma
+def U_C(gamma):
+    for edge in graph:
+        wire1 = edge[0]
+        wire2 = edge[1]
+        qml.CNOT(wires=[wire1, wire2])
+        qml.RZ(gamma, wires=wire2)
+        qml.CNOT(wires=[wire1, wire2])
+
+
+##############################################################################
+# With some foresight, we notice that we will need a way to sample
+# a measurement of multiple qubits in the computational basis, so we define
+# a Hermitian operator to do this. The eigenvalues of the operator are
+# the joint qubit measurement values in integer form.
+
+
+def comp_basis_measurement(wires):
+    n_wires = len(wires)
+    return qml.Hermitian(np.diag(range(2 ** n_wires)), wires=wires)
+
+
+##############################################################################
+# 2.4 Circuit
+# ~~~~~~~~~~~~~
+# Next, we create a quantum device with 4 qubits.
+
+dev1 = qml.device("default.qubit", wires=n_wires)
+
+##############################################################################
+# We also require a quantum node which will apply the operators :math:`p` times given the
+# angle parameters, and return the expectation value of the observable
+# :math:`\sigma_z^{j}\sigma_z^{k}` to be used in the term of the objective function later on. The
+# argument ``edge`` specifies the chosen edge term in the objective function, :math:`(j,k)`.
+#
+# Once optimized, the same quantum node can be used for sampling an approximately optimal bitstring
+# if executed with the ``edge`` keyword set to None.
+
+pauli_z = [[1, 0], [0, -1]]
+pauli_z_2 = np.kron(pauli_z, pauli_z)
+
+
+@qml.qnode(dev1)
+def circuit(params, edge=None, p=1):
+    # apply hadamards to get n qubit |+> state
+    for wire in range(n_wires):
+        qml.Hadamard(wires=wire)
+    # p instances of unitary operators
+    for i in range(p):
+        U_C(params[0][i])
+        U_B(params[1][i])
+    # during measurement phase we are evaluating
+    # the circuit with optimized parameters
+    if edge is None:
+        return qml.sample(comp_basis_measurement(range(n_wires)), n=1)
+    # during the optimization phase we are evaluating a term
+    # in the objective using exp val
+    return qml.expval(qml.Hermitian(pauli_z_2, wires=edge))
+
+
+##############################################################################
+# 2.5 Optimization
+# ~~~~~~~~~~~~~
+# Finally, we optimize the objective over the
+# angle parameters :math:`\boldsymbol{\gamma}` (``params[0]``) and :math:`\boldsymbol{\beta}`
+# (``params[1]``).
+# and then sample the optimized
+# circuit multiple times to yield a distribution of bitstrings. One of the optimal partitions
+# (z=0101 or z=1010) should be the most frequently sampled bitstring.
+
+
+def qaoa_maxcut(p=1):
+    print("\np=%d" % p)
+
+    # initialize the parameters near zero
+    init_params = 0.01 * np.random.rand(2, p)
+
+    # minimize negative the objective
```

Comment:
Lol, I was trying to convey the idea that in MaxCut we are trying to maximize an objective function F_p, which is the same thing as minimizing -1*F_p. I can change it to "minimize the objective"

Reply:
It may be useful to state this explicitly above when describing the optimization problem, i.e., define the optimization probelm as minimizing -C(z). You can explain this is done to follow the standard convention that optimization problems are cast as minimization

Reply:
Okay good idea